ariadne/contrib/federation/utils.py

```diff
@@ -66,14 +42,39 @@
 ]
 
 
+def _purge_directive_nodes(nodes: Tuple[Node, ...]) -> Tuple[Node, ...]:
+    return tuple(
+        node
+        for node in nodes
+        if not isinstance(node, (DirectiveNode, DirectiveDefinitionNode))
+        or node.name.value in _allowed_directives
+    )
+
+
+def _purge_type_directives(definition: Node):
+    # Recursively check every field defined on the Node definition
+    # and remove any directives found.
+    for key in definition.keys:
+        value = getattr(definition, key, None)
+        if isinstance(value, tuple):
+            # Remove directive nodes from the tuple
+            # e.g. doc -> definitions [DirectiveDefinitionNode]
+            next_value = _purge_directive_nodes(cast(Tuple[Node, ...], value))
+            for item in next_value:
+                if isinstance(item, Node):
+                    # Look for directive nodes on sub-nodes
+                    # e.g. doc -> definitions [ObjectTypeDefinitionNode] -> fields -> directives
+                    _purge_type_directives(item)
+            setattr(definition, key, next_value)
```

Comment:
Is this because `definition.key = next_value` annoyed linters?

Reply:
Yes that is part of it. The other reason I chose this path was to keep this flexible when traversing the tree of nodes so that future additions will be caught as well.
In order to use `definition.key` you must know all of the values of `key` upfront.